python/mxnet/model.py

```diff
@@ -122,10 +130,10 @@ def _initialize_kvstore(kvstore, param_arrays, arg_params, param_names, update_o
     """Initialize kvstore"""
     for idx, param_on_devs in enumerate(param_arrays):
         name = param_names[idx]
-        kvstore.init(name, arg_params[name])
-
-        if update_on_kvstore:
-            kvstore.pull(name, param_on_devs, priority=-idx)
+        if not update_on_kvstore or arg_params[name].stype != 'default':
```

Comment:
Where is the type 'default' defined? Also, it's better for readability to avoid using double negative logic.

Reply:
ok, i can update the branch condition. The stype is specified in ndarray documentation e.g. https://github.com/apache/incubator-mxnet/blob/master/python/mxnet/ndarray/utils.py#L51-L52 

default = dense
row_sparse = row_sparse
csr = compressed sparse row

Reply:
Okay, I guess I overlooked. The stype here is the storage type, not the type of kvstore.